src/Core/src/Platform/Android/KeyboardExtensions.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Android.Text;
+
+namespace Microsoft.Maui.Platform.Android
+{
+	public static class KeyboardExtensions
+	{
+		public static InputTypes ToInputType(this Keyboard self)
+		{
+			var result = new InputTypes();
+
+			// ClassText:																						!autocaps, spellcheck, suggestions 
+			// TextFlagNoSuggestions:																			!autocaps, !spellcheck, !suggestions
+			// InputTypes.ClassText | InputTypes.TextFlagCapSentences											 autocaps,	spellcheck,  suggestions
+			// InputTypes.ClassText | InputTypes.TextFlagCapSentences | InputTypes.TextFlagNoSuggestions;		 autocaps, !spellcheck, !suggestions
+
+			if (self == Keyboard.Default)
+				result = InputTypes.ClassText | InputTypes.TextVariationNormal;
+			else if (self == Keyboard.Chat)
+				result = InputTypes.ClassText | InputTypes.TextFlagCapSentences | InputTypes.TextFlagNoSuggestions;
+			else if (self == Keyboard.Email)
+				result = InputTypes.ClassText | InputTypes.TextVariationEmailAddress;
+			else if (self == Keyboard.Numeric)
+				result = InputTypes.ClassNumber | InputTypes.NumberFlagDecimal | InputTypes.NumberFlagSigned;
+			else if (self == Keyboard.Telephone)
+				result = InputTypes.ClassPhone;
+			else if (self == Keyboard.Text)
+				result = InputTypes.ClassText | InputTypes.TextFlagCapSentences;
+			else if (self == Keyboard.Url)
+				result = InputTypes.ClassText | InputTypes.TextVariationUri;
+			else if (self is CustomKeyboard)
+			{
+				var custom = (CustomKeyboard)self;
+				var capitalizedSentenceEnabled = (custom.Flags & KeyboardFlags.CapitalizeSentence) == KeyboardFlags.CapitalizeSentence;
+				var capitalizedWordsEnabled = (custom.Flags & KeyboardFlags.CapitalizeWord) == KeyboardFlags.CapitalizeWord;
+				var capitalizedCharacterEnabled = (custom.Flags & KeyboardFlags.CapitalizeCharacter) == KeyboardFlags.CapitalizeCharacter;
+
+				var spellcheckEnabled = (custom.Flags & KeyboardFlags.Spellcheck) == KeyboardFlags.Spellcheck;
+				var suggestionsEnabled = (custom.Flags & KeyboardFlags.Suggestions) == KeyboardFlags.Suggestions;
+
+				if (!capitalizedSentenceEnabled && !spellcheckEnabled && !suggestionsEnabled)
+					result = InputTypes.ClassText | InputTypes.TextFlagNoSuggestions;
+
+				if (!capitalizedSentenceEnabled && !spellcheckEnabled && suggestionsEnabled)
+				{
+					// Due to the nature of android, TextFlagAutoCorrect includes Spellcheck
+					// Log.Warning(null, "On Android, KeyboardFlags.Suggestions enables KeyboardFlags.Spellcheck as well due to a platform limitation.");
+					result = InputTypes.ClassText | InputTypes.TextFlagAutoCorrect;
+				}
+
+				if (!capitalizedSentenceEnabled && spellcheckEnabled && !suggestionsEnabled)
+					result = InputTypes.ClassText | InputTypes.TextFlagAutoComplete;
```

Comment:
Is it not possible to just `|` these together as we go along instead of checking for all the cases? Like:

```
if (capitalizedSentenceEnabled)
    result |= InputTypes.TextFlagCapSentences;
if (spellcheckEnabled)
    result |= InputTypes.TextFlagAutoCorrect;
```

Reply:
This is the exact copy from the renderer code. I did not make any enhancements to the code to be following PR guidelines :) Can make the changes in separate PR if needed.

Reply:
OK, no prob. Fixes can come later.